src/SDK/MonitoringApplianceEmulator/Controls/TracesControl.xaml

```diff
@@ -0,0 +1,112 @@
+                    </MultiBinding>
+                </ComboBox.IsEnabled>
+            </ComboBox>
+        </StackPanel>
+        
+        <DataGrid
+            Grid.Row="1" Grid.Column="0" Grid.ColumnSpan="2" x:Name="TracesGrid"
+            ItemsSource="{Binding PageableLog.CurrentPageTraces}" AutoGenerateColumns="False" SelectionMode="Single">
+            <DataGrid.ItemContainerStyle>
+                <Style TargetType="{x:Type DataGridRow}">
+                    <Style.Triggers>
+                        <DataTrigger Binding="{Binding Level}" Value="Verbose">
+                            <Setter Property="Foreground" Value="DarkGray"/>
+                        </DataTrigger>
+                        <DataTrigger Binding="{Binding Level}" Value="Warning">
+                            <Setter Property="Background" Value="Orange"/>
+                        </DataTrigger>
+                        <DataTrigger Binding="{Binding Level}" Value="Error">
+                            <Setter Property="Background" Value="Red"/>
+                            <Setter Property="Foreground" Value="White"/>
+                        </DataTrigger>
+                    </Style.Triggers>
+                </Style>
+            </DataGrid.ItemContainerStyle>
+            <DataGrid.Columns>
+                <DataGridTextColumn
+                    Header="Time Stamp" Binding="{Binding TimeStamp, StringFormat='yyyy-MM-ddTHH:mm:ss.fff'}"
+                    Width="SizeToCells" IsReadOnly="True" CanUserSort="False" CanUserReorder="False"/>
+                <DataGridTextColumn
+                    Header="Level" Binding="{Binding Level}"
+                    Width="SizeToCells" IsReadOnly="True" CanUserSort="False" CanUserReorder="False"/>
+                <DataGridTextColumn
+                    Header="Message" Binding="{Binding Message}"
+                    Width="SizeToCells" IsReadOnly="True" CanUserSort="False" CanUserReorder="False"/>
+            </DataGrid.Columns>
+        </DataGrid>
+
+        <StackPanel Orientation="Horizontal" HorizontalAlignment="Center" Grid.Row="2" Grid.Column="0" Grid.ColumnSpan="2">
+            <Button Style="{StaticResource TracesControlButton}" Command="{Binding FirstPageCommand}" IsEnabled="{Binding IsFirstPage, Converter={StaticResource InverseBooleanConverter}}">
+                <materialDesign:PackIcon Kind="ArrowCollapseLeft" VerticalAlignment="Center"/>
+            </Button>
+            <Button Style="{StaticResource TracesControlButton}" Command="{Binding PrevPageCommand}" IsEnabled="{Binding IsFirstPage, Converter={StaticResource InverseBooleanConverter}}">
+                <materialDesign:PackIcon Kind="ArrowLeft" VerticalAlignment="Center"/>
+            </Button>
+            <TextBlock Text="Page" VerticalAlignment="Center"/>
+            <TextBox Width="20" VerticalAlignment="Center" Margin="5,0" Text="{Binding CurrentPageIndex}"/>
```

Comment:
><TextBox Width="20" VerticalAlignment="Center" Margin="5,0" Text="{Binding CurrentPageIndex}"/> [](start = 12, length = 95)

Can the page number be typed here?

Reply:
yes, added improvement to accept also Enter

---
In reply to: [231003554](https://github.com/Azure/SmartDetectorsMonitoringAppliance/pull/13#discussion_r231003554) [](ancestors = 231003554)